### PR TITLE
PW-15: Image Lazy Loading

### PR DESCRIPTION
**Pull request by alexnatalia**
## Changes

* Update Element UI version.
* Added Skeleton component.

### CODE REVIEW

**Review by Nikita-Polyakov — APPROVED** [review]

File: src/components/Image/SImage/SImage.vue [hunk 1]
```diff
@@ -0,0 +1,81 @@
+<template>
+  <div class="s-image__container">
+    <s-skeleton
+      v-if="hasSkeleton"
+      :loading="!this.elImage || this.elImage.loading"
```
Comment (stefashkaa) [inline src/components/Image/SImage/SImage.vue:5]: `(elImage || {}).loading`

Comment (stefashkaa) [inline src/components/Image/SImage/SImage.vue:5]: please remove `this`, cuz it doesn't really matter in template section

File: src/components/Image/SImage/SImage.vue [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    ElImage,
+    SSkeleton,
+    SSkeletonItem
+  }
+})
+export default class SImage extends Vue {
+  /**
+   * Image source, same as native
+   */
+  @Prop({ default: null, type: String }) readonly src!: string | null
```
Comment (stefashkaa) [inline src/components/Image/SImage/SImage.vue:53]: it should be required as I see it, without default. Btw, `() => null`. You cannot set objects, arrays and null like this

File: src/components/Image/SImage/SImage.vue [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  /**
+   * Indicate how the image should be resized to fit its container, same as object-fit
+   */
+  @Prop({ default: ImageFit.NONE, type: String }) readonly fit?: string
```
Comment (stefashkaa) [inline src/components/Image/SImage/SImage.vue:57]: why did you use `?` ?

Comment (stefashkaa) [reply src/components/Image/SImage/SImage.vue:57]: the same for other properties

File: src/components/Image/SImage/SImage.vue [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import { ImageFit } from '../consts'
+
+@Component({
+  components: {
+    ElImage,
+    SSkeleton,
+    SSkeletonItem
+  }
+})
+export default class SImage extends Vue {
```
Comment (stefashkaa) [inline src/components/Image/SImage/SImage.vue:49]: Also, please add to every jsdoc block possible values, if you use custom enum set, and values which is set by default

File: src/styles/image.scss [hunk 5]
```diff
@@ -0,0 +1,5 @@
+.s-image {
+  &__container {
+    width: 100%;
+  }
+}
```
Comment (stefashkaa) [inline src/styles/image.scss:5]: add line break please

File: src/stories/SImage.stories.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    :z-index="zIndex"
+    :has-skeleton="hasSkeleton"
+    :animated="animated"
+  />`,
+  props: {
+    src: {
+      default: text('Src', 'https://picsum.photos/1024')
+    },
+    fit: {
+      default: select('Element', Object.values(ImageFit), ImageFit.NONE)
```
Comment (stefashkaa) [inline src/stories/SImage.stories.ts:27]: 'Fit'

**Review by Nikita-Polyakov — APPROVED** [review]

File: src/components/Image/SImage/SImage.vue [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  /**
+   * Image source, same as native
+   */
+  @Prop({ default: () => null, type: String }) readonly src!: string
```
Comment (stefashkaa) [inline src/components/Image/SImage/SImage.vue:38]: maybe `required`?

File: src/stories/SSkeleton.stories.ts [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  }
+})
+
+export const SkeletonItem = () => ({
+  components: {
+    SSkeleton,
+    SSkeletonItem
+  },
+  template: `<div class="s-flex" style="flex: 1; flex-direction: column;">
+              <s-skeleton animated :loading="true" style="width: 100%;">
```
Comment (stefashkaa) [inline src/stories/SSkeleton.stories.ts:71]: loading

File: src/stories/SImage.stories.ts [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    <s-image :src="imageSrc + '?random=15'" :lazy="true" style="height: inherit;" />
+  </div>`,
+  data: () => ({
+    imageSrc: 'https://picsum.photos/1024/300'
+  })
+})
+
+export const FailedImage = () => ({
+  components: { SImage },
+  template: '<s-image :src="null" style="height: 300px;" :lazy="true" />'
```
Comment (stefashkaa) [inline src/stories/SImage.stories.ts:74]: src="''"

Comment (stefashkaa) [reply src/stories/SImage.stories.ts:74]: lazy

File: src/stories/SImage.stories.ts [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    animated: {
+      default: boolean('Skeleton has Animation', true)
+    }
+  }
+})
+
+export const LazyImages = () => ({
+  components: { SImage },
+  template: `<div style="width: 100%; overflow-y: auto; height: 300px;">
+    <s-image :src="imageSrc + '?random=1'" :lazy="true" style="height: inherit;" />
```
Comment (stefashkaa) [inline src/stories/SImage.stories.ts:51]: lazy

File: src/components/Image/SImage/SImage.vue [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+  @Prop({ default: '', type: String }) readonly alt!: string
+  /**
+   * Whether to use lazy load
+   */
+  @Prop({ default: false, type: Boolean }) readonly lazy!: boolean
+  /**
+   * The container to add scroll listener when using lazy load.
+   * Posssible value is the nearest parent container whose overflow property is auto or scroll
+   */
+  @Prop({ default: () => null, type: [String, HTMLElement] }) readonly scrollContainer!: string | HTMLElement
```
Comment (stefashkaa) [inline src/components/Image/SImage/SImage.vue:57]: remove default: () => null please

**Review by stefashkaa — APPROVED** [review]

**Pull request merged**